EngineLayer/GlycoSearch/GlycoSearchEngine.cs

```diff
@@ -0,0 +1,516 @@
+﻿using EngineLayer.ModernSearch;
+using MzLibUtil;
+using Proteomics;
+using Proteomics.Fragmentation;
+using Proteomics.ProteolyticDigestion;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EngineLayer;
+using MassSpectrometry;
+
+namespace EngineLayer.GlycoSearch
+{
+    public class GlycoSearchEngine : ModernSearchEngine
+    {
+        protected readonly List<GlycoSpectralMatch>[] GlobalCsms;
+
+        private bool IsOGlycoSearch;
+        private readonly int TopN;
+        private readonly int _maxOGlycanNum;
+        private readonly bool OxoniumIonFilt;
```

Comment:
try not to abbreviate variable names, it is confusing. I assume this abbreviation means "OxoniumIonFilter".  I don't know what that means, i know you're filtering oxonium ions somehow but i don't know how specifically

Reply:
fixed and comment added.